### ISSUE TITLE
Define Task Manager functionality (again)

### ISSUE THREAD

**Issue by adriatic** [monterey-framework/monterey#318]
### Intention

**I'm submitting a feature request**  

**This discussion should have a direct positive impact to [TaskManager: improve the UI](https://github.com/monterey-framework/monterey/issues/287)**

**Comment by JeroenVinke** [comment]
Currently the taskmanager has two large functions: to show the progress of running tasks, and to start new tasks (also called the "task runner" functionality).

Initially the taskmanager could do only the former (show progress of running tasks). It was implemented as a dialog so that it could be brought up regardless of where you were in Monterey (because the button to open the taskmanager dialog is in the taskbar). 

The task runner functionality was added much later, because there had to be a way to start new tasks (like gulp watch, au run). There are tasks that run only for a few seconds, and you may want to run these tasks repeatedly. It's handy if you can monitor and start tasks from the same screen. The question is whether we can live with starting tasks from tiles on the main screen and immediately opening the taskmanager so you see the output of the command. More on this below.

![image](https://cloud.githubusercontent.com/assets/2189477/18609883/1c0aa3b4-7d0d-11e6-9b9f-5f38272cb503.png)

We can easily extract the Workflows tab into a tile + screen, so we don't need to worry about that. 

Based on what project you have selected, Monterey adds several tabs. In the following screenshot you see "Skeleton-Esnext" which is a gulp project, therefore a gulp tab is shown with all gulp commands that Monterey knows of:
![image](https://cloud.githubusercontent.com/assets/2189477/18609886/3fa77b80-7d0d-11e6-99a5-b2c643842b4b.png). If you would have selected an aurelia-cli project, then you would have an equivalent `aurelia-cli` tab with `au run` and `au run --watch`.

But you don't want to go through the entire list of gulp commands in order to start `gulp watch`. You only use a limited amount of commands often, a problem which was solved by marking some commands as favorite.

The favorite tab shows all commands that have been marked as favorite:

![image](https://cloud.githubusercontent.com/assets/2189477/18609899/b561b584-7d0d-11e6-9b91-4a5f05a4f617.png)

Here we have a second set of tiles (the first set of tiles is in the main screen, outside of the taskmanager dialog). So there is a very strong resemblance between what is shown in the taskmanager, and what is shown in the main screen, where there is a list of projects at the left side of the screen and tiles at the right side of the screen:
![image](https://cloud.githubusercontent.com/assets/2189477/18609901/c07f3766-7d0d-11e6-82f4-26e2c522906e.png)

So the taskmanager has quite a few features right now:

- Monitor execution of commands
- Manage favorite commands
- Run favorite commands (as tiles for convenience)
- Create/edit/remove workflows
- Run gulp/aurelia-cli/webpack commands

We can reduce the complexity of the task manager a lot by moving all features except the first one (monitoring of execution of commands) to the main screen, where we have a tile + screen infrastructure that we can extend very easily

![image](https://cloud.githubusercontent.com/assets/2189477/18609974/5ddbf7aa-7d0f-11e6-965f-bae7d0f21073.png)

Assuming that we can live with starting new tasks as tiles from the main screen and then immediately bringing up the taskmanager for monitoring the progress. But I haven't thought beyond the point of assigning tiles to categories so the user can add a bunch of commands into a category without the problem of having too many tiles

**Comment by JeroenVinke** [comment]
I'd like to add that I find the "Run" tile very powerful, because I don't have to open the task manager and start a command from there. So can we do something like this for **all** tasks?

**Comment by adriatic** [comment]
___I want to immediately acknowledge my happiness with the quality and quantity of the information you provided in this issue (issue being the GitHub discussion thread we are in now). I will write more as I get through your text.___

**Comment by adriatic** [comment]
### Keeping the minimum data in the Task manager list (left side of the view)

> The task runner functionality was added much later, because there had to be a way to start new tasks (like gulp watch, au run)

I believe that my **[item 1](https://github.com/monterey-framework/monterey/wiki/My-views-on-the-definition-of-the-Monterey-Task-manager-functionality#1-keep-only-application-in-the-task-manager-left-side-navigation-bar)** and **[item 2](https://github.com/monterey-framework/monterey/wiki/My-views-on-the-definition-of-the-Monterey-Task-manager-functionality#2-remove-action-items-like--start-new-task-from-the-task-manager-left-side-navigation-bar)** may have a large impact on your thoughts stated after the above quote, so I would like to wait for your comments to this specific proposal before proceeding further.

**Comment by JeroenVinke** [comment]
I proposed the same, where new tasks are started from the main screen and not the taskmanager. But you should be able start any command like `gulp prepare-release`

**Comment by JeroenVinke** [comment]
Item 2 suggests that in order to start something like `gulp watch` for inactive projects, you would first have to click on the run tile, while running the app is not what you want

**Comment by adriatic** [comment]
My basic premise is **not to have inactive projects listed** - not only that  these do not belong here, but rather their absence changes the scope for global buttons in the Task manager

**Comment by adriatic** [comment]
> I proposed the same, where new tasks are started from the main screen and not the taskmanager. 

Yes, I realized that we are in complete sync only after starting to write the huge detailed descriptions that you clearly do not need. If we were in the same room, that would not happen of course and I started this big text, burned million times having to deal with a lot less competent folks 😄

**Comment by adriatic** [comment]
> We can reduce the complexity of the task manager a lot by moving all features except the first one (monitoring of execution of commands) to the main screen, where we have a tile + screen infrastructure that we can extend very easily

I do not think this is needed, once we agree that the left sidebar has only running apps (clearly that could be more than a single app - think about Rob using Monterey). So, the Task manager "local buttons" on top of the right side panel would apply to the currently selected running app in the left sidebar.

**Comment by JeroenVinke** [comment]
Yes it is a tricky one, which is why I have implemented improvements to our current implementation: https://github.com/monterey-framework/monterey/issues/287#issuecomment-247836309
